src/ORM/Query.php

```diff
@@ -287,7 +287,7 @@ public function eagerLoader(EagerLoader $instance = null)
      */
     public function contain($associations = null, $override = false)
     {
-        if (empty($associations) && $override) {
+        if ($override) {
             $this->_eagerLoader = null;
```

Comment:
I hadn't considered that, that behavior happened before when `contain(null, true)` was used in the past. I see the concern that `contain($thing, true)` will wipe matching() calls though.